### PR TITLE
S2 display teams t6

### PR DESCRIPTION
**Pull request by hb020202**

### CODE REVIEW

File: src/index.php [hunk 1]
```diff
@@ -1,4 +1,47 @@
 <?php
 
-require_once '../vendor/autoload.php';
+//require_once '../vendor/autoload.php';
```
Comment (jjtapley) [inline src/index.php]: uncomment this and delete other requires

File: vendor/autoload.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
```
Comment (jjtapley) [inline vendor/autoload.php:1]: I think all vendor files need to go into gitignore file?

File: src/dbConn.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+namespace TheRealMVP;
+use PDO;
+
+/**Connects to database
+ *
+ * No params given
+ *
+ * @return PDO
+ */
+class dbConn
```
Comment (jjtapley) [inline src/dbConn.php]: I think this needs to not be a class, but the function needs to be a public function inside the TeamHydrator class?

File: src/dbGetData.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+use PDO;
+
+
+/**Pulls info from the database
+ *
+ * @param $connection & $query
+ *
+ * @return string & integer
+ */
+class dbGetData
```
Comment (jjtapley) [inline src/dbGetData.php]: Again, I think this function needs to be inside the TeamHydrator class? And delete the dbGetData class? 

Then, the TeamHydrator class with connect to the DB, get the info out, and hydrate the Team objects?

File: src/dbConn.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?php
+namespace TheRealMVP;
+use PDO;
```
Comment (jjtapley) [inline src/dbConn.php]: Don't think you need use PDO?

File: src/dbConn.php (hunk 3 above)
Comment (jjtapley) [reply src/dbConn.php]: Also, I'm fairly sure that the connection is what Rich and Elmi are working on, so in theory we should be able to use part of their code to connect to DB and then pass that connection in when doing your getData request? So we may not need this function once Rich and Elmi have finished.

File: src/index.php [hunk 6]
```diff
@@ -1,4 +1,13 @@
 <?php
-
+namespace TheRealMVP;
 require_once '../vendor/autoload.php';
 
+
```
Comment (jjtapley) [inline src/index.php]: I don't think this should just be in index.php. Think the query should be inside the getData function?

File: vendor/autoload.php (hunk 2 above)
Comment (richcatt) [reply vendor/autoload.php:1]: Hmm, I think the composer files are OK to include...

File: src/dbGetData.php (hunk 4 above)
Comment (hb020202) [reply src/dbGetData.php]: But classes are only meant to do one thing, we thought? That's why we've kept them separate.

File: src/dbConn.php (hunk 3 above)
Comment (richcatt) [reply src/dbConn.php]: Hmm, we should probably create another class if we want to share the connection in multiple places...

File: src/TeamHydrator.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+use PDO;
+
+
+class TeamHydrator {
+    /**
+     * create database connection and retrieves data and returns an array of team objects
+     *
+     * @return array
+     */
+    public static function getData()
```
Comment (richcatt) [inline src/TeamHydrator.php]: Can we have some return type hinting please? :)

File: index.php [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?php
+require 'vendor/autoload.php';
+use TheRealMVP\TeamHydrator;
+
+$data = TeamHydrator::getData();
```
Comment (richcatt) [inline index.php]: FYI don't need to use 'use' if only referencing the class once - can just write inline

Fine though either way

File: src/TeamHydrator.php [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<?php
+namespace TheRealMVP;
+use PDO;
```
Comment (richcatt) [inline src/TeamHydrator.php]: Can just reference \PDO inline if you like

**Review by richcatt — CHANGES_REQUESTED** [review]

File: src/dbConn.php (hunk 3 above)
Comment (richcatt) [reply src/dbConn.php]: As discussed, leave for now

File: src/index.php (hunk 6 above)
Comment (richcatt) [reply src/index.php]: Think this is OK... Ultimately you'll need to link into index.

File: vendor/autoload.php (hunk 2 above)
Comment (richcatt) [reply vendor/autoload.php:1]: Ash just confirmed these shouldn't be included but leave for now as Ming and Ben have added to .gitignore

**Pull request merged**